File: ThirdYear/prakhal-gupta_PrakhalGupta_2125IT1045_3/Week_1/SetMatrixZeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m  = matrix.size();
        int n = matrix[0].size();
        unordered_set<int> row,col;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j]==0){
                    row.insert(i);
                    col.insert(j);
                }
            }
        }
        
        for(auto r:row){
            for(int j=0;j<n;j++)
                matrix[r][j]=0;
        }
        
        for(auto c:col){
            for(int i=0;i<m;i++)
                matrix[i][c]=0;
        }   
    }
};

File: ThirdYear/amaan-lari_AmaanLari_2125it1039_3/Week_1/SetMatrixZeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int rows = matrix.length;
        int columns = matrix[0].length;
        boolean isCol = false;

        for(int i = 0; i < rows; i++) {
            if(matrix[i][0] == 0) {
                isCol = true;
            }
            for(int j = 1; j < columns; j++) {
                 if(matrix[i][j] == 0) {
                    matrix[i][0] = 0;
                    matrix[0][j] = 0;
                }
            }
        }

        for(int i = 1; i < rows; i++) {
            for(int j = 1; j < columns; j++) {
                if(matrix[i][0] == 0 || matrix[0][j] == 0) {
                    matrix[i][j] = 0;
                }
            }
        }

        if(matrix[0][0] == 0) {
            for(int j = 0; j < columns; j++) {
                matrix[0][j] = 0;
            }
        }

        if(isCol) {
            for(int i = 0; i < rows; i++) {
                matrix[i][0] = 0;
            }
        }
    }
}